includes/Updater.php

```diff
@@ -0,0 +1,195 @@
+<?php
+/**
+ * Class Updater
+ *
+ * @package   Google\Web_Stories
+ * @copyright 2020 Google LLC
+ * @license   https://www.apache.org/licenses/LICENSE-2.0 Apache License 2.0
+ * @link      https://github.com/google/web-stories-wp
+ */
+
+/**
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Google\Web_Stories;
+
+/**
+ * Plugin updater class.
+ *
+ * Allows updating from beta releases to the most current version.
+ *
+ * @todo Remove for stable release.
+ */
+class Updater {
+	/**
+	 * Registers functionality through WordPress hooks.
+	 *
+	 * @return void
+	 */
+	public function init() {
+		add_filter(
+			'plugins_api',
+			function( $value, $action, $args ) {
+				return $this->plugin_info( $value, $action, $args );
+			},
+			10,
+			3
+		);
+		add_filter(
+			'pre_set_site_transient_update_plugins',
+			function( $value ) {
+				return $this->updater_data( $value );
+			}
+		);
+		add_action(
+			'load-update-core.php',
+			function() {
+				$this->clear_plugin_data();
+			}
+		);
+		add_action(
+			'upgrader_process_complete',
+			function( $upgrader, $options ) {
+				if (
+					'update' !== $options['action'] ||
+					'plugin' !== $options['type'] ||
+					! isset( $options['plugins'] ) ||
+					! in_array( plugin_basename( WEBSTORIES_PLUGIN_FILE ), $options['plugins'], true )
+				) {
+					return;
+				}
+				$this->clear_plugin_data();
+			},
+			10,
+			2
+		);
+	}
+
+	/**
+	 * Retrieves plugin information data from custom API endpoint.
+	 *
+	 * @param false|object|array $value  The result object or array. Default false.
+	 * @param string             $action The type of information being requested from the Plugin Installation API.
+	 * @param mixed              $args   Plugin API arguments.
+	 * @return false|object|array Updated $value, or passed-through $value on failure.
+	 */
+	private function plugin_info( $value, $action, $args ) {
```

Comment:
Why is this private?

Reply:
I just realized I broke this by applying your suggestions.

It needs to be public when directly used for `add_action`...

I assume it was private to not unnecessarily expose it. I just copied the code.